utils/llm.py

```diff
@@ -144,7 +144,8 @@ def route_llm_prompt(text_chunk, instructions, source_langage, target_language):
     body = {
         "source_language": source_langage,
         "target_language": target_language,
-        "prompt": [{"role": "system", "content": instructions}, {"role": "user", "content": text_chunk}]
+        "messages": [{"role": "system", "content": instructions}, {"role": "user", "content": text_chunk}],
+        "llm_providers": llm_providers
     }
     response = requests.post(
```

Comment:
The response unpacking is incorrect. You need to do `select_model = response.json().get("provider").get("model")`